src/ia/js/IAPage.js

```diff
@@ -167,25 +167,58 @@
-                            var max_height = 0;
-                            $(".dev_milestone-container").each(function(idx) {
-                                if ($(this).height() > max_height) {
-                                    max_height = $(this).height();
-                                }
-                            });
-
-                            $(".dev_milestone-container").height(max_height);
+                            // Hide the faux-window if the screenshot doesn't exist.
+                            $(".ia-single--screenshots").hide();
+                            $(".ia-single--right").addClass("dashed-border");
 
                             page.imgHide = true;
                         }
                     });
+
+                    // Saves the screenshot to S3.
+                    $(".save-screenshot--button").click(function() {
+                        $.post("/screenshot/save/" + ia_data.live.id, function(data) {
+                            // Hide the save button.
+                            $(".save-screenshot--button").addClass("hide");
+                            // Put in the image from S3.
+                            $(".ia-single--image-container img").attr("src", "https://images.duckduckgo.com/iu/?u=" + encodeURIComponent(data.screenshots.index));
+                        });
+                    });
+
+                    // Generate a screenshot when the button is clicked.
+                    $(".generate-screenshot--button").click(function() {
+                        $(".generate-screenshot--button").html("Generating ...");
+
+                        // Send a POST request with the ID of the IA.
+                        $.post("/screenshot/create/" + ia_data.live.id, function(data) {
+                            // Check if the screenshot that we want is available.
+                            // If it isn't there must be something wrong.
+                            if(data.screenshots.index) {
+                                $(".generate-screenshot--button").html("Generate Screenshot");
+                                $(".save-screenshot--button").removeClass("hide");
+
+                                // Show preview image.
+                                $(".ia-single--image-container img").attr("src", data.screenshots.index);
+                                $(".ia-single--screenshots").show();
+                                $(".ia-single--right").removeClass("dashed-border");
+                            } else {
+                                $(".generate-screenshot--button").html("Generate Screenshot");
+                            }
+                        }).error(function() {
+                            // Revert button to its original state since the server is down.
+                            // Add a little delay.
+                            setTimeout(function() {
+                                $(".generate-screenshot--button").html("Generate Screenshot");
```

Comment:
Maybe print a message error above the button, too? "Ooops, something went wrong - try again!" (just an idea)